### ISSUE TITLE
Class-specific FDR in OpenMS 

### ISSUE THREAD

**Issue by ypriverol** [OpenMS/OpenMS#5339]
Hi @timosachsenberg and @all : 

We have been using OpenMS and proteomicsLFQ to perform LFQ-proteogenomics analysis of public proteomics data. We have used and stringent FDR at the peptide level to assess the quality of all our peptides., however "global" FDR for the entire database is not probably the best way to estimate the peptide FDR in proteogenomics studies. 

We will need in the `IDFilter` tool a possibility to compute the class-specific FDR which is a more stringent FDR on top of the global FDR that enable to filter peptides of the same class for example: `variant peptides` or `pseudogene peptides`, etc. 

Best reference to the topic here: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4392723/  

This issue aims to trigger the discussion about how to implement this feature in IDFilter and what are the basics of class-specific FDR. 

I'm tagging some people who I think are interested in this topic @timosachsenberg @jpfeuffer @husensofteng

**Comment by timosachsenberg** [comment]
I think the main challenge defining the different classes we are interested in and check if we have
all information to annotate the proteins with class labels.
E.g., for pseudogenes it seems that it is often annotated in the description:
>tr|A0A3B3IT33|A0A3B3IT33_HUMAN Tripartite motif-containing 51G, pseudogene OS=Homo sapiens OX=9606 GN=TRIM51GP PE=4 SV=1
Though I don't know how reliable this information is.
Once this is annotated we would need to figure out how to transfer that information to peptides/PSMs (e.g., if they are not unique).
Once we have the class labels transferred to peptides, filtering should be easy.

**Comment by ypriverol** [comment]
I think this is an "standard" we can agree in the same way we define decoy prefix. The main problem is that some part of the sequence are shared between the canonical protein and the pseudo-genes, then some peptides will be identified in both the canonical and the specific class. Then, it is a canonical peptide and not a pseudogene.

**Comment by timosachsenberg** [comment]
1. so for unique peptides we are done - we just take the annotation
2. for shared peptides it is a bit more complicated.
2.1 if a shared peptide maps to only pseudogenes I think we could assign the pseudogene class label
2.2 if a shared peptide maps to both pseudogene and non-pseudogene we probably don't want to assign the pseudogene class label (or do we e.g., if the protein probability of the non-pseudogene would be super low?)

@jpfeuffer does this interact with the inference?

**Comment by husensofteng** [comment]
Exactly, the prefix has to be set as a rule and the priority for shared peptides need to be pre-defined but of course the canonical class always has the highest priority.

Another concern is to ensure that the number of decoys per class is sufficient for class-specific FDR but we could leave that for that the user to decide. The effect of this is to make the classes dynamic.

**Comment by husensofteng** [comment]
The suffix in the accession sounds good. We could provide that when generating the DB.

I think the ordered list of classes would allow for more flexibility without re-generating the DB.

One additional point is to allow for combined classes in the ordered list. 
example:
canonical, pseudogene&lncRNA, variant. 
in such case the peptides from pseudogene and lncRNA should be treated as one class.

**Comment by timosachsenberg** [comment]
One additional note: To make the calculation of class-specific FDR more generic and also compatible with e.g., calculating FDRs of PSMs with certain modifications we could probably annotate a class label at the PSM level (e.g., based on the suffix and transfer rules or the modification state).
This should be compatible with what has been suggested here and would work for both proteogenomics annotations and e.g., modification discovery.
